rollup/sync_service.go

```diff
@@ -1,255 +1,139 @@
+	syncing           bool
+	LatestL1ToL2      LatestL1ToL2
+	confirmationDepth uint64
 }
 
 // NewSyncService returns an initialized sync service
 func NewSyncService(ctx context.Context, cfg Config, txpool *core.TxPool, bc *core.BlockChain, db ethdb.Database) (*SyncService, error) {
-	if txpool == nil {
-		return nil, errors.New("Must pass TxPool to SyncService")
-	}
 	if bc == nil {
 		return nil, errors.New("Must pass BlockChain to SyncService")
 	}
 
 	ctx, cancel := context.WithCancel(ctx)
 	_ = cancel // satisfy govet
 
-	if cfg.TxIngestionSignerKey == nil {
-		cfg.TxIngestionSignerKey, _ = crypto.GenerateKey()
-	}
-	address := crypto.PubkeyToAddress(cfg.TxIngestionSignerKey.PublicKey)
-
 	if cfg.IsVerifier {
 		log.Info("Running in verifier mode")
+	} else {
+		log.Info("Running in sequencer mode")
 	}
 
-	// Layer 2 chainid to use for signing
+	// Layer 2 chainid
 	chainID := bc.Config().ChainID
+	// Initialize the rollup client
+	client := NewClient(cfg.RollupClientHttp, chainID)
 	service := SyncService{
-		ctx:                              ctx,
-		cancel:                           cancel,
-		verifier:                         cfg.IsVerifier,
-		enable:                           cfg.Eth1SyncServiceEnable,
-		heads:                            make(chan *types.Header),
-		doneProcessing:                   make(chan uint64),
-		eth1HTTPEndpoint:                 cfg.Eth1HTTPEndpoint,
-		AddressResolverAddress:           cfg.AddressResolverAddress,
-		CanonicalTransactionChainAddress: cfg.CanonicalTransactionChainAddress,
-		SequencerDecompressionAddress:    cfg.SequencerDecompressionAddress,
-		L1CrossDomainMessengerAddress:    cfg.L1CrossDomainMessengerAddress,
-		confirmationDepth:                cfg.Eth1ConfirmationDepth,
-		signer:                           types.NewOVMSigner(chainID),
-		key:                              *cfg.TxIngestionSignerKey,
-		address:                          address,
-		gasLimit:                         cfg.GasLimit,
-		txpool:                           txpool,
-		bc:                               bc,
-		eth1ChainId:                      cfg.Eth1ChainId,
-		eth1NetworkId:                    cfg.Eth1NetworkId,
-		ctcDeployHeight:                  cfg.CanonicalTransactionChainDeployHeight,
-		db:                               db,
-		clearTransactionsAfter:           (5760 * 18), // 18 days worth of blocks
-		clearTransactionsTicker:          time.NewTicker(time.Hour),
-		sequencerIngestTicker:            time.NewTicker(15 * time.Second),
-		txCache:                          NewTransactionCache(),
-		HeaderCache:                      [headerCacheSize]*types.Header{},
+		ctx:               ctx,
+		cancel:            cancel,
+		verifier:          cfg.IsVerifier,
+		enable:            cfg.Eth1SyncServiceEnable,
+		confirmationDepth: cfg.Eth1ConfirmationDepth,
+		bc:                bc,
+		txpool:            txpool,
+		eth1ChainId:       cfg.Eth1ChainId,
+		client:            client,
+		db:                db,
+	}
+
```

Comment:
Note that the data-transport service indexes things at the ctc index, so it starts at index 0 instead of 1. The current sync service here will start syncing from index 1 instead of 0, meaning that it misses a single transaction. Be sure to update this to start syncing from 0